### DISCUSSION TITLE
Should operator= and copy constructor work the same for JSON object

### DISCUSSION THREAD

**Discussion by billpriest** [nlohmann/json#3905] [Q&A]
All,
  The "lint" program I used suggested changing from
auto obj = myObj.at("key");

to

auto obj{myObj.at("key")};

I expected these to work identically.  Is this reasonable to expect?
I tried v3.11.2 and the former works and the latter doesn't using g++ 11.3.0 on Linux x86.
If needed I can provide a working/not working test case.

**Comment by nlohmann** [comment]
No, they are behaving differently. See #2583.

**Comment by billpriest** [comment]
Thanks Niels for the response and the awesome json "library".  I should have tried w/ clang; I have builds for both. Seems a "Catch 22"; I was starting to believe that compiler bugs like this were a thing of the past.  Not sure why my search didn't find the issue sited; sorry for the noise.

Bill

________________________________
From: Niels Lohmann ***@***.***>
Sent: Saturday, December 31, 2022 1:04 AM
To: nlohmann/json ***@***.***>
Cc: Bill Priest ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL]Re: [nlohmann/json] Should operator= and copy constructor work the same for JSON object (Discussion #3905)


No, they are behaving differently. See #2583<https://github.com/nlohmann/json/issues/2583>.
